### ISSUE TITLE
Server Side Generation error - window is not defined

### ISSUE THREAD

**Issue by nicolopadovan** [portabletext/editor#419]
Might be a framework specific issue, hence label this issue as you find most appropriate.
However, using the [example](https://github.com/portabletext/editor/blob/main/examples/basic/src/App.tsx) - adding the `use client` directive as well - there is an issue related to `window` not being defined (are you missing a `useEffect`?).

I will try to look into this problem myself as well as soon as I free up some of my time :D 

```
 ⨯ node_modules/.pnpm/@portabletext+editor@1.8.0_@sanity+block-tools@3.64.1_@sanity+schema@3.64.1_@sanity+types@3.6_mrzwfjnau5pqahjr3bryqb6rfe/node_modules/@portabletext/editor/lib/index.mjs (5602:1) @ eval
 ⨯ uncaughtException: ReferenceError: window is not defined
    at Array.forEach (<anonymous>)
    at renderElement (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:4815:25)
    at renderElement (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5805:21)
    at retryNode (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5631:10)
    at renderNodeDestructive (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:6181:17)
    at renderNode (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:6066:8)
    at renderChildrenArray (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5818:12)
    at retryNode (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5631:10)
    at renderNodeDestructive (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5297:12)
    at renderElement (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5805:21)
    at retryNode (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5631:10)
    at renderNodeDestructive (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:4766:12)
    at finishFunctionComponent (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5208:10)
    at renderElement (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5805:21)
    at retryNode (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5631:10)
    at renderNodeDestructive (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:6181:17)
    at renderNode (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:6066:8)
    at renderChildrenArray (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5818:12)
    at retryNode (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5631:10)
    at renderNodeDestructive (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:6181:17)
    at renderNode (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5246:10)
    at renderElement (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5805:21)
    at retryNode (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5631:10)
    at renderNodeDestructive (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5535:14)
    at renderElement (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5805:21)
    at retryNode (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5631:10)
    at renderNodeDestructive (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:4766:12)
    at finishFunctionComponent (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5208:10)
    at renderElement (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5805:21)
    at retryNode (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5631:10)
    at renderNodeDestructive (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:6181:17)
    at renderNode (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:6066:8)
    at renderChildrenArray (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5818:12)
    at retryNode (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5631:10)
    at renderNodeDestructive (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:6181:17)
    at renderNode (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5246:10)
    at renderElement (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5805:21)
    at retryNode (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5631:10)
    at renderNodeDestructive (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:4766:12)
    at finishFunctionComponent (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5208:10)
    at renderElement (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5805:21)
    at retryNode (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:5631:10)
    at renderNodeDestructive (webpack://next/dist/compiled/react-dom/cjs/react-dom-server.edge.development.js:4766:12)
  5600 |     });
  5601 |   };
> 5602 |   return window.addEventListener("online", onlineHandler), window.addEventListener("offline", offlineHandler), () => {
       | ^
  5603 |     window.removeEventListener("online", onlineHandler), window.removeEventListener("offline", offlineHandler);
  5604 |   };
  5605 | }), editorMachine = setup({
```

**Comment by christianhg** [comment]
Thank you for reporting this. I'll soon remove this piece of code, but until then, I hope this fix resolves the issue: #429 429

**Comment by christianhg** [comment]
Update: I've now removed this code completely, and thus the problem you are experiencing here will most definitely be gone if you upgrade to the latest version, `1.10.2`.
